Applications/Spire/Source/OrderImbalanceIndicator/OrderImbalanceIndicatorContextMenu.cpp

```diff
@@ -0,0 +1,197 @@
+    const ToggledSignal::slot_type& slot) const {
+  return m_side_signal.connect(slot);
+}
+
+connection OrderImbalanceIndicatorContextMenu::connect_size_toggled_signal(
+    const ToggledSignal::slot_type& slot) const {
+  return m_size_signal.connect(slot);
+}
+
+connection
+    OrderImbalanceIndicatorContextMenu::connect_reference_price_toggled_signal(
+    const ToggledSignal::slot_type& slot) const {
+  return m_ref_px_signal.connect(slot);
+}
+
+connection OrderImbalanceIndicatorContextMenu::connect_date_toggled_signal(
+    const ToggledSignal::slot_type& slot) const {
+  return m_date_signal.connect(slot);
+}
+
+connection OrderImbalanceIndicatorContextMenu::connect_time_toggled_signal(
+    const ToggledSignal::slot_type& slot) const {
+  return m_time_signal.connect(slot);
+}
+
+connection
+    OrderImbalanceIndicatorContextMenu::connect_notional_value_toggled_signal(
+    const ToggledSignal::slot_type& slot) const {
+  return m_notional_value_signal.connect(slot);
+}
+
+bool OrderImbalanceIndicatorContextMenu::eventFilter(QObject* watched,
+    QEvent* event) {
+  if(watched == m_table_columns_menu) {
+    if(event->type() == QEvent::Show) {
+      m_table_columns_menu->move(pos().x() + scale_width(136),
+        pos().y() + scale_height(6));
+    }
+  }
+  return QMenu::eventFilter(watched, event);
+}
+
+void OrderImbalanceIndicatorContextMenu::add_check_box(const QString& text,
+    ToggledSignal& signal) {
+  auto font = QFont("Roboto");
+  font.setPixelSize(scale_height(12));
+  auto metrics = QFontMetrics(font);
+  auto spacing = scale_width(108) - metrics.horizontalAdvance(text);
+  auto text_style = QString(R"(
+    color: black;
+    font-family: %4;
+    font-size: %1px;
+    outline: none;
+    padding-right: %3px;
+    spacing: %2px;
+    text-align: left center;)")
+    .arg(font.pixelSize()).arg(spacing).arg(scale_width(8))
+    .arg(font.family());
+  auto indicator_style = QString(R"(
+    background-color: white;
+    border: %1px solid #C8C8C8 %2px solid #C8C8C8;
+    height: %3px;
+    width: %4px;)").arg(scale_height(1))
+    .arg(scale_width(1)).arg(scale_height(16)).arg(scale_width(16));
+  auto checked_style = QString(R"(
+    image: url(:/Icons/check-with-box.svg);)");
+  auto hover_style = QString(R"(
+    border: %1px solid #4B23A0 %2px solid #4B23A0;)")
+    .arg(scale_height(1)).arg(scale_width(1));
+  auto focused_style = QString(R"(border-color: #4B23A0;)");
+  auto check_box = new CheckBox(text, this);
+  check_box->setFixedSize(CHECK_BOX_SIZE());
+  check_box->setChecked(true);
+  check_box->set_stylesheet(text_style, indicator_style, checked_style,
+    hover_style, focused_style);
+  check_box->setLayoutDirection(Qt::RightToLeft);
+  auto action = new QWidgetAction(this);
+  action->setDefaultWidget(check_box);
+  m_table_columns_menu->addAction(action);
+  connect(static_cast<QCheckBox*>(action->defaultWidget()),
+    &QCheckBox::stateChanged, [&] (int state) {
```

Comment:
Never capture by reference if the closure will outlive the scope of its declaration!

Reply:
You will have to do something annoying like:

    [signal = &signal] (int state) {
      (*signal)(...);
    }

What this does is capture a pointer to the existing ```signal``` by value.